internal/server/v1/dlq/handler_test.go

```diff
@@ -0,0 +1,231 @@
+package dlq
```

Comment:
let's use `dlq_test` for test package, reason is, we don't want the `dlq` package to share any private variables with the test, in this case, the `dlq` package will have access to `mock*` structs that are here.


Secondly, if your reason for this is because handler methods are private then a few approaches can be: 1. make handler's methods public or 2. test the router, not the handler.

Personally I would just test whatever the package is exposing, and in this current setup, Router is the output of this package, handler is the implementation details, hence I would test the router instead. example: https://github.com/goto/dex/blob/main/internal/server/v1/alert/routes_test.go#L37

Reply:
hm. in golang we write tests in the same package. let's discuss this?

Reply:
if you are doing whitebox testing then yes.
But most of the times, we only want to test exported API from the package we are testing. (e.g. blackbox testing)

Using different package will make sure:
1. test can only access exported APIs which simulates how the package is being used by other packages using it
2. no variables conflict - you can create any vars or const to help refactor your test without having to interfere with application package (e.g. `dlq`)

So basically there is no hard rule per se, you can still use same package, but I don't see any benefits doing that unless I want to have extensive tests on private APIs. (e.g. testing complex logic but the API is not exposed)